storage/storage-client/src/lib.rs

```diff
@@ -275,6 +282,52 @@ impl StorageRead for StorageReadServiceClient {
             .boxed();
         Ok(stream)
     }
+
+    // TODO migrate this to `ConfigStorage` trait as non-async
+    // and implement for LibraDBTrait once `StorageRead` is deprecated
+    async fn batch_fetch_config(&self, access_paths: Vec<AccessPath>) -> Vec<Option<Vec<u8>>> {
```

Comment:
It seems to me this api can be used not only for configs but also for arbitrary move value. Maybe rename the function? cc @msmouse

Reply:
If storage experts can confirm there will be a use case for this other than reading configs, I can update this name

Reply:
I think this logic was used in a bunch of locations. Currently to fetch a resource on chain, we just do a get_account_state from the client and perform the rest of the logic locally to get the resource they are interested in, and this is a pattern that occurred in multiple locations of the codebase. We might want to have a unified story of reading Move values on chain @davidiw . With that being said I'm ok with proceeding with this api as long as that unblocks you and the storage folks are ok about it.